source/docs/one-command-deployment.md

```diff
@@ -43,18 +47,17 @@ deploy:
   message: [message] #leave this blank
 ```
 
-| Option    | Description                                                                                                 |
-| --------- | ----------------------------------------------------------------------------------------------------------- |
-| `repo`    | GitHub/Bitbucket/Coding/GitLab repository URL                                                               |
-| `branch`  | Branch name. The deployer will detect the branch automatically if you are using GitHub or GitCafe.          |
-| `message` | Customize commit message (Default to `Site updated: {% raw %}{{ now('YYYY-MM-DD HH:mm:ss') }}{% endraw %}`) |
-
-3. Upload your site: `./node_modules/.bin/hexo clean && ./node_modules/.bin/hexo deploy` (or `hexo clean && hexo deploy` if you installed Hexo globally).
-4. On Github/BitBucket/Gitlab go to your repository settings and change your main branch from `master` to `published` (or whatever you called it in your \_config.yml). Now your site will show as your account's homepage.
-
-### How does it work exactly?
+| Option    | Description |
+| --------- | --- |
+| `repo`    | URL of the target repository |
+| `branch`  | Branch name. Defaults to `gh-pages` on GitHub and `master` for others |
+| `message` | Customize commit message. Defaults to `Site updated: {% raw %}{{ now('YYYY-MM-DD HH:mm:ss') }}{% endraw %}` |
+| `token`   | Optional token value to authenticate with the repo. Prefix with `$` to read token from environment variable |
 
-Your repository will have a **master** branch when you first made it. Keep working on this branch to create your site. When you deploy Hexo will create, or update, a new branch on the remote site (called **published** in the config above). Deployment won't create a new branch locally, nor will it mess with your existing source code in the master branch locally or on the remote. You still need to keep pushing commits to the master branch manually to the remote server to keep your site backed up. In addition, if you are using a CNAME file to customize your Github Pages domain name, you need to put the CNAME file under `source_dir` so that Hexo can push it to the published branch.
+3. Deploy your site, `./node_modules/.bin/hexo clean && ./node_modules/.bin/hexo deploy` (or `hexo clean && hexo deploy` if you installed Hexo globally).
+  - You will be prompted with username and password of the target repository, unless you authenticate with a token.
+  - hexo-deployer-git does not store your username and password. Use [git-credential-cache](https://git-scm.com/docs/git-credential-cache) to store them temporarily.
+4. Navigate to your repository settings and change the "Pages" branch to `published` (or the branch specified in your config). The deployed site should be live on the link shown on the "Pages" setting.
```

Comment:
GitHub only support publish site to GitHub Pages from:

- root directory of gh-pages branch
- root directory of master branch
- docs directory of master branch

And by default, hexo-deployer-git use those branches if no `branch` option is given:

- GitHub.com: gh-pages branch
- Coding.net: coding-pages branch
- Others: master branch

Reply:
Add default branches